src/package-resolver.js

```diff
@@ -420,6 +427,26 @@ export default class PackageResolver {
     return maxValidRangeManifest;
   }
 
+  exoticRangeMatch(resolvedPkgs: Iterable<Manifest>, manifest: Manifest): ?Manifest {
+    if (!manifest._remote) {
+      return null;
+    } else if (manifest._remote.type != 'copy' || !manifest._remote.reference) {
```

Comment:
You return `null` in both cases so why not:
```js
if (!(manifest._remote && manifest._remote.reference && manifest._remote.type === 'copy')) {
    return null;
}
```

This also makes your intent clearer.